### PR TITLE
Clock Overflow

### PR DESCRIPTION
**Pull request by coffee-cup**
This PR handled the system clock overflow for periodic tasks

## Closes
- #28

### CODE REVIEW

**Comment by coffee-cup** [comment]
Confirmed it works for the 8 lights using an 8bit clock. Not sure if we want to put this in our report as we don't have tests for it. Will hopefully be useful when we use the RTOS for project 3 though.

**Comment by zevisert** [comment]
Might be able to work it as something the tests discovered?

File: project2/kernel.c [hunk 1]
```diff
@@ -204,6 +207,7 @@ void Kernel_Task_Create() {
                 Process[x].wcet = request_info->wcet;
                 Process[x].ttns = sys_clock + request_info->offset;
                 Process[x].ticks_remaining = Process[x].wcet;
+                Process[x].clockState = clock_state;
```
Comment (zevisert) [inline project2/kernel.c]: Good, this means that even if a process is created waaaaay late into the program's execution, the descriptor's clock state will be correct  👍

**Comment by coffee-cup** [comment]
@zevisert  yeah good point!

**Pull request merged**